src/parse-templates.ts

```diff
@@ -286,89 +203,13 @@ export function parseTemplates(
    * literal is top level (e.g. not nested within a dynamic segment) and has a
    * tag, pushes it into the list of results.
    */
-  function parseTemplateLiteral(
-    results: TemplateMatch[],
-    template: string,
-    startToken: RegExpMatchArray,
-    tokens: RegExpMatchArray[],
-    isTopLevel = false,
-    importedNames: Map<string, StaticImportConfig>
-  ) {
-    let hasDynamicSegment = false;
-
-    while (tokens.length > 0) {
-      let currentToken = expect(tokens.shift(), 'expected token');
-
-      if (isEscaped(template, currentToken.index)) continue;
-
-      if (currentToken[0].match(dynamicSegmentStart)) {
-        hasDynamicSegment = true;
-
-        parseDynamicSegment(results, template, currentToken, tokens);
-      } else if (currentToken[0].match(templateLiteralEnd)) {
-        if (isTopLevel && !hasDynamicSegment && startToken[1]?.length > 0) {
-          // Handle the case where a tag-like was matched, e.g. hbs`hello`
-          if (currentToken[0].length > 1) {
-            const tokenStr = currentToken[0];
-            const index = expect(currentToken.index, 'expected index');
-
-            currentToken = ['`'];
-            currentToken.index = index + tokenStr.length - 1;
-          }
-          const tagName = startToken[1];
-          const importConfig = importedNames.get(tagName);
-
-          if (importConfig !== undefined) {
-            let contents = '';
-
-            if (startToken.index !== undefined) {
-              const templateStart = startToken.index + startToken[0].length;
-
-              contents = template.slice(templateStart, currentToken.index);
-            }
-
-            results.push({
-              type: 'template-literal',
-              tagName,
-              contents: contents,
-              start: startToken,
-              end: currentToken,
-              importPath: importConfig.importPath,
-              importIdentifier: importConfig.importIdentifier,
-            });
-          }
-        }
-
-        return;
-      }
-    }
-  }
-
-  /**
-   * Parses a dynamic segment within a template literal. Continues parsing until
-   * the dynamic segment has been exited, ignoring all tokens within it.
-   * Accounts for nested block statements, strings, and template literals.
-   */
-  function parseDynamicSegment(
-    results: TemplateMatch[],
-    template: string,
-    _startToken: RegExpMatchArray,
-    tokens: RegExpMatchArray[]
-  ) {
-    let stack = 1;
-
+  function parseTemplateLiteral(template: string, tokens: RegExpMatchArray[]) {
```

Comment:
I think this code predates anyone who still looks at this project, so I could just be misunderstanding, but why do we still need this function at all? Isn't "template literal" the term the implementation here uses to refer to the tagged-string version of templates that you're removing with this PR?

Reply:
Template tags may exist with actual strings, such as within a REPL that demonstrates how to build a REPl with itself. anything in a string, we want to ignore.

Reply:
So this is now doing the opposite of its name and _avoiding_ accidentally parsing string content as a template?

Reply:
exactly, yeah -- the function was pulling double duty previously